circuits/plonk-plookup/src/gates/packing.rs

```diff
@@ -0,0 +1,48 @@
+/*****************************************************************************************************************
+
+This source file implements packing constraint gate Plonk primitive.
+
+PACK gate constrains
+    si = s1,i + 2*s2,i + 4*s3,i + 8*s4,i + 16*si+1
+    s1,i * (s1,i – 1) = 0
+
+*****************************************************************************************************************/
+
+use algebra::FftField;
+use crate::wires::{GateWires, COLUMNS};
+use crate::gate::{CircuitGate, GateType};
+use array_init::array_init;
+
+impl<F: FftField> CircuitGate<F>
+{
+    pub fn create_pack(row: usize, wires: GateWires) -> Self
+    {
+        CircuitGate
+        {
+            row,
+            typ: GateType::Pack,
+            wires,
+            c: vec![]
+        }
+    }
+
+    pub fn verify_pack(&self, witness: &[Vec<F>; COLUMNS]) -> bool
+    {
+        let this: [F; COLUMNS] = array_init(|i| witness[i][self.row]);
+        let next: [F; COLUMNS] = array_init(|i| witness[i][self.row+1]);
+
+        self.typ == GateType::Pack
+        &&
+        next[4] ==
```

Comment:
I guess it doesn't really make a difference, but is there a reason it's more convenient to split the bits across the 2 rows in this way?